src/org/edx/hadoop/input/ManifestTextInputFormat.java

```diff
@@ -0,0 +1,77 @@
+package org.edx.hadoop.input;
+
+import java.io.*;
+import java.util.List;
+import java.util.ArrayList;
+
+import org.apache.hadoop.fs.FileStatus;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.mapred.JobConf;
+import org.apache.hadoop.mapred.KeyValueTextInputFormat;
+
+
+public class ManifestTextInputFormat extends KeyValueTextInputFormat {
+    public static int RETRY_LIMIT = 5;
+
+    protected FileStatus[] listStatus(JobConf job) throws IOException {
+        FileStatus[] manifests = super.listStatus(job);
+        List<FileStatus> paths = new ArrayList<FileStatus>();
+        for(int i = 0; i < manifests.length; i++) {
+            List<Path> globPaths = this.readManifest(manifests[i].getPath(), job);
+            for (Path globPath : globPaths) {
+                if (globPath != null) {
+                    FileStatus fs = getFileStatus(globPath, job);
+                    if (fs != null)
+                        paths.add(fs);
+                }
+            }
+        }
+        return paths.toArray(new FileStatus[1]);
+    }
+
+    private List<Path> readManifest(Path manifestPath, JobConf job) throws IOException {
+        FileSystem fs = manifestPath.getFileSystem(job);
+        List<Path> paths = new ArrayList<Path>();
+        DataInputStream dataStream = fs.open(manifestPath);
+        try {
+            BufferedReader reader = new BufferedReader(new InputStreamReader(dataStream));
+            String line = reader.readLine();
+            while(line != null) {
+                paths.add(new Path(line));
+                line = reader.readLine();
+            }
+        } finally {
+            dataStream.close();
+        }
+
+        return paths;
+    }
+
+    private FileStatus getFileStatus(Path targetPath, JobConf conf) throws IOException {
+        return getFileStatusWithRetry(targetPath, conf, RETRY_LIMIT);
+    }
+
+    private FileStatus getFileStatusWithRetry(Path targetPath, JobConf conf, int retryCount) throws IOException {
+        if (retryCount <= 0) {
+            LOG.info("Retries exhausted, dropping file: " + targetPath.toUri());
```

Comment:
Correct, this was the initial path of failure that got circumvented by the throwing of the IOException.  It will still exist as a safeguard however